packages/flutter/lib/src/painting/shader_warm_up.dart

```diff
@@ -16,10 +16,24 @@ import 'package:flutter/foundation.dart';
 ///
 /// Therefore, we use this during the [PaintingBinding.initInstances] call to
 /// move common shader compilations from animation time to startup time. By
-/// default, a [DefaultShaderWarmUp] is used. Create a custom [ShaderWarmUp]
-/// subclass to replace [PaintingBinding.shaderWarmUp] before
-/// [PaintingBinding.initInstances] is called. Usually, that can be done before
-/// calling [runApp].
+/// default, a [DefaultShaderWarmUp] is used. If needed, create a custom
+/// [ShaderWarmUp] subclass and give it to [PaintingBinding.shaderWarmUp]
+/// (so it replaces [DefaultShaderWarmUp]) before
+/// [PaintingBinding.initInstances] is called. Usually, that can be done
+/// before calling [runApp].
+///
+/// To determine whether a draw operation is useful for warming up shaders,
+/// check the difference in the worst_frame_rasterizer_time_millis benchmarks.
+/// Also, tracing with `flutter run --profile --trace-skia` could reveal if
+/// there are shader compilation related janks. If there are, some long
+/// `GrGLProgramBuilder::finalize` calls would appear in the middle of an
+/// animation. Their parent calls which look like `XyzOp` would suggest Xyz
+/// draw operations are causing the shaders to be compiled. A useful shader
+/// warm-up draw operation would eliminate such long compilation calls in the
+/// animation. To double check the warm-up, trace with `flutter run --profile
+/// --trace-skia --start-paused`. The `GrGLProgramBuilder` with the associated
+/// `XyzOp` should appear during the startup rather than in the middle of a
```

Comment:
Is XyzOp a hypothetical name, like FooOp, or is it an actual thing?

Reply:
It's a hypothetical name.

Reply:
Fixed in doc updates.

Reply:
@cbracken updated. Thank you very much for the help!